api/src/planDeTransport/ligneDeBus/ligneDeBusService.ts

```diff
@@ -178,35 +207,80 @@ export const updatePDRForLine = async (
   );
 
   if (shouldSendEmailCampaign && youngUpdateResult.modifiedCount > 0) {
-    await sendEmailCampaign(ligneBusId, newMeetingPointId, cohort);
+    const updatedYoungs = await YoungModel.find({ ligneId: ligneBusId, meetingPointId: newMeetingPointId });
+    await notifyYoungsAndRlsPDRWasUpdated(updatedYoungs, cohort);
   }
 
   return ligneBus;
 };
 
-const sendEmailCampaign = async (ligneBusId: string, newMeetingPointId: string, cohort: CohortType) => {
-  const updatedYoungs = await YoungModel.find({
-    ligneId: ligneBusId,
-    meetingPointId: newMeetingPointId,
-  }).select("email parent1Email parent2Email");
-
-  let templateId: string | null = null;
-  if (new Date() < new Date(cohort.dateStart)) {
-    templateId = SENDINBLUE_TEMPLATES.young.CHANGE_PDR_BEFORE_DEPARTURE;
-  } else if (new Date() < new Date(cohort.dateEnd)) {
-    templateId = SENDINBLUE_TEMPLATES.young.CHANGE_PDR_BEFORE_RETURN;
+export const updateSessionForLine = async ({
+  ligne,
+  session,
+  user,
+  sendCampaign,
+}: {
+  ligne: LigneBusDocument;
+  session: SessionPhase1Document;
+  user: UserDto;
+  sendCampaign?: boolean;
+}) => {
+  const currentSessionPhase1 = await SessionPhase1Model.findById(ligne.sessionId);
+  if (!currentSessionPhase1) throw new Error(ERRORS.NOT_FOUND);
+
+  if (ligne.youngSeatsTaken > (session.placesLeft || 0)) {
+    throw new Error(ERRORS.OPERATION_NOT_ALLOWED);
   }
 
-  if (!templateId) throw new Error("Modification date is out of range, no email sent.");
-
-  for (const young of updatedYoungs) {
-    const cc: { email: string }[] = [];
-    if (young.parent1Email) cc.push({ email: young.parent1Email });
-    if (young.parent2Email) cc.push({ email: young.parent2Email });
-
-    await sendTemplate(templateId, {
-      emailTo: [{ email: young.email }],
-      cc,
+  const transaction = await startSession();
+
+  try {
+    await withTransaction(transaction, async () => {
+      // Ligne
+      ligne.set({
+        sessionId: session.id,
+        centerId: session.cohesionCenterId,
+      });
+      await ligne.save({ fromUser: user });
+
+      // Plan de transport
+      const planDeTransport = await PlanTransportModel.findById(ligne._id);
+      if (!planDeTransport) throw new Error(ERRORS.NOT_FOUND);
+      planDeTransport.set({
+        centerId: session.cohesionCenterId,
+        centerRegion: session.region,
+        centerDepartment: session.department,
+        centerAddress: session.cityCentre,
+        centerZip: session.zipCentre,
+        centerName: session.nameCentre,
+        centerCode: session.codeCentre,
+      });
+      await planDeTransport.save({ fromUser: user });
+
+      // Jeunes
+      const filter = { ligneId: ligne._id };
+      const updateDoc = { $set: { sessionPhase1Id: session.id, cohesionCenterId: session.cohesionCenterId } };
+      await YoungModel.updateMany(filter, updateDoc, { fromUser: user });
```

Comment:
il faut passer la transaction à chaque save ou update :
```suggestion
      await YoungModel.updateMany(filter, updateDoc, { fromUser: user, session: transaction });
```